test/hooks/permissionedPools/PermissionedV4Router.t.sol

```diff
@@ -0,0 +1,862 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {Currency, CurrencyLibrary} from "@uniswap/v4-core/src/types/Currency.sol";
+import {IV4Router} from "../../../src/interfaces/IV4Router.sol";
+import {PermissionedRoutingTestHelpers} from "./shared/PermissionedRoutingTestHelpers.sol";
+import {Planner} from "../../shared/Planner.sol";
+import {Actions} from "../../../src/libraries/Actions.sol";
+import {ActionConstants} from "../../../src/libraries/ActionConstants.sol";
+import {IAllowlistChecker} from "../../../src/hooks/permissionedPools/interfaces/IAllowlistChecker.sol";
+import {WrappedPermissionedToken} from "../../../src/hooks/permissionedPools/WrappedPermissionedToken.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IHooks} from "@uniswap/v4-core/src/interfaces/IHooks.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {MockPermissionedToken} from "./PermissionedPoolsBase.sol";
+
+contract PermissionedV4RouterTest is PermissionedRoutingTestHelpers {
```

Comment:
can you add a test where the permissioned token is in the middle of a 3 hop swap? so non permissioned => permissioned => non permissioned, this would mean the user wouldn't have to transfer a permissioned token but the allowlist should still be verified on the middle swap